test/device/common.hpp

```diff
@@ -35,7 +35,18 @@ namespace rocwmma
     template <typename T>
     __device__ inline float64_t toDouble(T const& val)
     {
-        return static_cast<float64_t>(static_cast<float32_t>(val));
+        using rocwmma::float32_t;
+        using rocwmma::float64_t;
+
+        // Small floating point types need to convert up to single before double.
+        if constexpr((sizeof(T) < sizeof(float32_t)) && std::is_floating_point_v<T>)
```

Comment:
same as hiptensor. 

I noticed that hip does not define is_floating_point_v for bf16

`std::is_floating_point_v<bf16>  == false`
`std::is_floating_point_v<_Float16 >  == true`